### PR TITLE
Option to skip unknown characters

### PR DESCRIPTION
**Pull request by emlyn**
It would be nice to have an option to skip unknown characters instead of replacing them with a default. This PR enables you to do that by setting `notfound=None`.

Especially if #178 is merged, then all standard alphanumeric characters will have some representation, and I think it would make sense to make it the default? Users can still get the old behaviour by setting `notfound='_'`.

While there, I also simplified the loops to use `for`/`in` instead of explicitly dealing with iterators, and fixed (what i think is) a bug, in that `dot_muncher` was ignoring any dot at the start of the string.

This PR would need a bit of tweaking if #178 is merged, but I preferred to base it on top of master for now so that the changes are clearer.

### CODE REVIEW

**Comment by emlyn** [comment]
Looks like the coverage decrease is just due to device.py getting reduced in size so the non-covered lines are now a higher fraction of the total; all the lines I touched are covered.

**Comment by rm-hull** [comment]
Yep, reduced code in the PR seems much cleaner. The only doubt I have is whether the default value should be “_” rather than None, if only to retain backwards compatibility. What do you think? 

@thijstriemstra - do you have an opinion about the default value?

**Comment by thijstriemstra** [comment]
I'm for backwards compatibility @rm-hull

File: luma/led_matrix/segment_mapper.py [hunk 1]
```diff
@@ -76,34 +76,29 @@
...
-        iterator = iter(text)
-        while True:
-            char = next(iterator)
-            yield _DIGITS.get(char, undefined)
-    except StopIteration:
-        pass
+def regular(text, notfound=None):
+    undefined = _DIGITS[notfound] if notfound is not None else None
+    for char in iter(text):
+        d = _DIGITS.get(char, undefined)
```
Comment (thijstriemstra) [inline luma/led_matrix/segment_mapper.py]: can a more descriptive var name be used instead of `d`?

**Comment by emlyn** [comment]
Thanks for looking at this. I agree that it makes sense to keep the default backwards compatible (especially without #178 merged).

**Pull request merged**